ccm_web/server/appHandler.ts

```diff
@@ -0,0 +1,145 @@
+import path from 'path'
+
+import { NextHandleFunction } from 'connect'
+import express from 'express'
+import type { Express, Request, Response, Router } from 'express'
+import { IdToken, Provider } from 'ltijs'
+import Database from 'ltijs-sequelize'
+import webpack from 'webpack'
+import webpackDevMiddleware from 'webpack-dev-middleware'
+import type { WebpackDevMiddleware } from 'webpack-dev-middleware'
+
+import { Config } from './config'
+import devWebpackConfig from '../webpack/webpack.dev'
+
+interface DevOptions {
+  isDev: true
+}
+
+interface ProdOptions {
+  isDev: false
+  staticPath: string
+}
+
+interface WebpackDevOutput {
+  middleware: WebpackDevMiddleware & NextHandleFunction
+  publicPath: string
+}
+
+class AppHandler {
+  private readonly config: Config
+  private readonly apiRouter: Router
+  private readonly envOptions: DevOptions | ProdOptions
+
+  constructor (
+    config: Config,
+    envOptions: DevOptions | ProdOptions,
+    apiRouter: Router
+  ) {
+    this.config = config
+    this.envOptions = envOptions
+    this.apiRouter = apiRouter
+  }
+
+  static handleProdClient (app: Express, staticPath: string): void {
+    console.log('Loading dist/client...')
+    app.use(express.static(staticPath))
+
+    app.get('*', (req: Request, res: Response) => {
+      res.sendFile(path.join(staticPath, 'index.html'))
+    })
+  }
+
+  /* TO DO: Investigate some way to dynamically import webpack stuff? */
+  static setupWebpackMiddleware (): WebpackDevOutput {
+    console.log('Setting up webpack-dev-middleware...')
+
+    const publicPath = devWebpackConfig.output?.publicPath
+    if (publicPath === undefined) throw Error('Webpack publicPath was not properly defined!')
+
+    const compiler = webpack(devWebpackConfig)
+    const webpackDevMid = webpackDevMiddleware(compiler, {
+      publicPath: String(publicPath),
+      // Just copies what's used in memory to client/build (changing doesn't do anything)
+      writeToDisk: true
+    })
+    return { middleware: webpackDevMid, publicPath: String(publicPath) }
+  }
+
+  // ltijs docs: https://cvmcosta.me/ltijs/#/
+  async setupLTI (): Promise<Express> {
```

Comment:
This Block of `setLTI` console.log doesn't work. even I enable `DEBUG=provider:*`. 
I wanted to know what is in side Token.
`provider.onConnect(async (token: IdToken, req: Request, res: Response) => {`

How did you determine the token is `IdToken`?

Reply:
I don't know what's going on with this honestly. I will take a look. I believe it's `IdToken` based on the source code and the `@types` updates that the contributor proposed.

Reply:
So you can see the token logged if you use the production workflow offered through `docker-compose-prod.yml`. So maybe something `provider.redirect` is causing an issue here?

Reply:
I now am wondering if this callback is getting disrupted by the Webpack Middleware. I'll focus on this tomorrow morning, try to dig into how this interacts with ltijs.

Reply:
I'm fairly confident now the problem is the Webpack middleware, but I don't have a solution yet. 😞

Reply:
Probably may be tied with how webpack is restarting the server when BE changes happened. How exactly is webpack interfering here?